tests/tt_eager/profiling/profile_host_overhead.py

```diff
@@ -0,0 +1,294 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import pytest
+import torch
+import tt_lib
+import time
+import statistics
+from loguru import logger
+from models.utility_functions import torch2tt_tensor
+from tests.tt_eager.profiling import ops_for_profiling
+from tracy import signpost
+
+
+test_sweep_args = [
+    # (
+    #     (1, 2, 1024, 1024),
+    #     tt_lib.tensor.DataType.BFLOAT16,
+    #     tt_lib.tensor.Layout.TILE,
+    #     tt_lib.tensor.MemoryConfig(tt_lib.tensor.TensorMemoryLayout.INTERLEAVED, tt_lib.tensor.BufferType.DRAM),
+    #     tt_lib.tensor.MemoryConfig(tt_lib.tensor.TensorMemoryLayout.INTERLEAVED, tt_lib.tensor.BufferType.DRAM),
+    # ),
+    (
+        (1, 4, 1024, 1024),
+        tt_lib.tensor.DataType.BFLOAT16,
+        tt_lib.tensor.Layout.TILE,
+        tt_lib.tensor.MemoryConfig(tt_lib.tensor.TensorMemoryLayout.INTERLEAVED, tt_lib.tensor.BufferType.DRAM),
+        tt_lib.tensor.MemoryConfig(tt_lib.tensor.TensorMemoryLayout.INTERLEAVED, tt_lib.tensor.BufferType.DRAM),
+    ),
+]
+
+all_num_call_to_stack = [1, 3]  # For 10 and more test  execution spills to dispatch
+num_repeats = 5
+
+
+def measure_host_overhead(op_func, op_name, device, num_call_to_stack, is_warmup):
+    if not is_warmup:
```

Comment:
Making sure I understood this function correctly, op total time is (calling n times + sync)/n time and overhead time  is just (calling n times)/n? 

I would say the names are a little misleading. Overhead suggests that if we remove it, total time will drop by that amount. However, because of how we are calling n times, depending on how much kernels are taking on device, overhead might go to zero with total not decreasing by that amount. This is because kernel execution is started while you are still calling the ops from host side.